packages/credentials/src/status-list-credential.ts

```diff
@@ -0,0 +1,238 @@
+ *
+ */
+export class StatusListCredential {
+  /**
+   * Create a [StatusListCredential] with a specific purpose, e.g., for revocation.
+   *
+   * @param statusListCredentialId The id used for the resolvable path to the status list credential [String].
+   * @param issuer The issuer URI of the credential, as a [String].
+   * @param statusPurpose The status purpose of the status list cred, eg: revocation, as a [StatusPurpose].
+   * @param issuedCredentials The credentials to be included in the status list credential, eg: revoked credentials, list of type [VerifiableCredential].
+   * @returns A special [VerifiableCredential] instance that is a StatusListCredential.
+   * @throws Error If the status list credential cannot be created.
+   *
+   * Example:
+   * ```
+      StatusListCredential.create({
+        statusListCredentialId : 'https://statuslistcred.com/123',
+        issuer                 : issuerDid.uri,
+        statusPurpose          : StatusPurpose.REVOCATION,
+        issuedCredentials      : [credWithCredStatus]
+      })
+   * ```
+   */
+  public static create(options: StatusListCredentialCreateOptions): VerifiableCredential {
+    const { statusListCredentialId, issuer, statusPurpose, issuedCredentials } = options;
+    const statusListIndexes: string[] = this.prepareCredentialsForStatusList(statusPurpose, issuedCredentials);
+    const bitString = this.bitstringGeneration(statusListIndexes);
+
+    const credentialSubject = {
+      id            : statusListCredentialId,
+      type          : 'StatusList2021',
+      statusPurpose : statusPurpose,
+      encodedList   : bitString,
+    };
+
+    const vcDataModel: VcDataModel = {
+      '@context'        : [DEFAULT_VC_CONTEXT, DEFAULT_STATUS_LIST_VC_CONTEXT],
+      type              : [DEFAULT_VC_TYPE, DEFAULT_STATUS_LIST_VC_TYPE],
+      id                : statusListCredentialId,
+      issuer            : issuer,
+      issuanceDate      : getCurrentXmlSchema112Timestamp(),
+      credentialSubject : credentialSubject,
+    };
+
+    return new VerifiableCredential(vcDataModel);
+  }
+
+  /**
+   * Validates if a given credential is part of the status list represented by a [VerifiableCredential].
+   *
+   * @param credentialToValidate The [VerifiableCredential] to be validated against the status list.
+   * @param statusListCredential The [VerifiableCredential] representing the status list.
+   * @returns A [Boolean] indicating whether the `credentialToValidate` is part of the status list.
+   *
+   * This function checks if the given `credentialToValidate`'s status list index is present in the expanded status list derived from the `statusListCredential`.
+   *
+   * Example:
+   * ```
+   * const isRevoked = StatusListCredential.validateCredentialInStatusList(credentialToCheck, statusListCred);
+   * ```
+   */
+  public static validateCredentialInStatusList(
```

Comment:
Design: there is probably a combination of misunderstanding of the spec, bad naming of the method, or bad ChatGPT documentation generation 😆 :

When the value is is 1 in the given index position, it does not mean the credential is "in the status list" as the method name and documentation states. It just means the credential is revoked. So we ought to rename the method to something like `isCredentialRevoked()`, though I generally prefer having method checking for the happy case, in which case `isCredentialValid()` or `validateCredentialStatus()`.

Reply:
I agree it needs to be updated along with the other api naming changes, This is the same naming as web5-kt and yea will want to do a light design and update the api naming across all sdks

A problem here is that the function name has to be generic because it can be revoked, suspended, annulled, withdrawn ...

But yes will add to the issue